src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -75,8 +82,9 @@ private void Initialize(string clusterName, string databaseName, string azureAcc
             _kustoAdminProvider = KustoClientFactory.CreateCslAdminProvider(stringBuilder);
         }
 
-        public void UpdateAzureToken(string azureAccountToken)
+        private void RefreshAzureToken()
         {
+            string azureAccountToken = ConnectionService.Instance.RefreshAzureToken(_ownerUri);
```

Comment:
There is too much threading of this parameter through all the classes which do not need it. I am thinking if we can avoid it in some manner. The ConnectionService has all the ConnectionInfos. The ConnectionInfo has the list of active ReliableDataSources->DataSource. I wonder if we can just call RefreshAzureToken and pass it a reference to the IDataSource we are calling from, and ConnectionService can match the ConnectionInfo it needs to operate on.